kubernetes-discovery-client/src/main/java/io/micronaut/kubernetes/discovery/informer/InstanceProviderInformerNamespaceResolver.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.discovery.informer;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.kubernetes.KubernetesConfiguration;
+import io.micronaut.kubernetes.discovery.KubernetesServiceConfiguration;
+import io.micronaut.kubernetes.discovery.KubernetesServiceInstanceProvider;
+import jakarta.inject.Singleton;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Resolver for the namespaces to watch by {@link KubernetesServiceInstanceProvider}.
+ *
+ * @author Pavol Gressa
+ * @since 3.2
+ */
+@Internal
+@Singleton
+public class InstanceProviderInformerNamespaceResolver {
```

Comment:
When the watch mode is enabled we need to find out what namespaces to create the Informer for. That's the responsibility of this resolver. It resolves the namespaces by the manually configured service discovery and the mode.